### ISSUE TITLE
Any auth option for the url endpoint?

### ISSUE THREAD

**Issue by ninodafonte** [jdbranham/grafana-diagram#154]
Hi Jeremy!

First of all, thank you for this plugin, it's a great tool that I'm actively using in Grafana.

I was wondering if you have (or had) any plans for including any type of auth for the endpoint gathering the graph definition.

If that's something that you think makes sense, I could give it a try at least to include some configuration (user/pass) and go for basic auth maybe ... 

Looking forward for your comments about it, 

Thank you!

**Comment by jdbranham** [comment]
Thanks! 
Auth hasn't been mentioned before, but as long as we don't introduce any breaking changes, I'd gladly merge new functionality if you create a PR!
Let me know if you run into any issues.

**Comment by ninodafonte** [comment]
Great! I'm taking care of it. 
I've added some properties to the diagramControl and changed the get request, looking good so far. 

I've noticed the layout for the first part of the editor (content, url, custom styles, etc) looked kind  of "broken" so I'm using the opportunity to refactor some stuff there as well (not a front-end but I think it will be ok ;)), hope you don't mind. 

Hopefully in a couple of days you'll get some news
